tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -68,26 +70,66 @@ def notEmpty(start=0, end=None):
 
 # custom validators
 
-def validate_single_header_trailer(file):
+def validate_single_header_trailer(datafile):
     """Validate that a raw datafile has one trailer and one footer."""
+    line_number = 0
     headers = 0
     trailers = 0
+    is_valid = True
+    error_message = None
 
-    for rawline in file:
+    for rawline in datafile.file:
         line = rawline.decode()
+        line_number += 1
 
         if line.startswith('HEADER'):
             headers += 1
         elif line.startswith('TRAILER'):
             trailers += 1
 
         if headers > 1:
-            return (False, 'Multiple headers found.')
+            is_valid = False
+            error_message = 'Multiple headers found.'
+            break
 
         if trailers > 1:
-            return (False, 'Multiple trailers found.')
+            is_valid = False
+            error_message = 'Multiple trailers found.'
+            break
 
     if headers == 0:
-        return (False, 'No headers found.')
-
-    return (True, None)
+        is_valid = False
+        error_message = 'No headers found.'
+
+    error = None
+    if not is_valid:
+        error = generate_parser_error(
+            datafile=datafile,
+            line_number=line_number,
+            schema=None,
+            error_category=ParserErrorCategoryChoices.PRE_CHECK,
+            error_message=error_message,
+            record=None,
+            field=None
+        )
+
+    return is_valid, error
+
+
+def validate_header_section_matches_submission(datafile, section):
+    """Validate header section matches submission section."""
+    is_valid = datafile.section == section
+
+    error = None
+    if not is_valid:
+        error = generate_parser_error(
+            datafile=datafile,
+            line_number=1,
+            schema=None,
+            error_category=ParserErrorCategoryChoices.PRE_CHECK,
+            error_message="Section does not match.",
```

Comment:
would more info in the error message be helpful here @reitermb? such as what is mentioned in the docstring of this function?